src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -0,0 +1,74 @@
+# sourcery skip: use-fstring-for-concatenation
+
+from lxml import etree
+
+from dsp_tools.commands.xmlupload.ontology_client import OntologyClient, format_ontology
+from dsp_tools.commands.xmlupload.ontology_diagnose_models import OntoCheckTool, Ontology, UnknownOntologyElements
+from dsp_tools.models.exceptions import BaseError
+
+
+def do_xml_consistency_check(onto_client: OntologyClient, root: etree._Element) -> None:
+    """
+    This function takes an OntologyClient and the root of an XML.
+    It retrieves the ontologies from the server.
+    It iterates over the root.
+    If it finds any unknown properties or classes, they are printed out and a UserError is raised.
+
+     Args:
+         onto_client: client for the ontology retrieval
+         root: root of the XML
+
+     Raises:
+         UserError: if there are any unknown properties or classes
+    """
+    onto_tool = OntoCheckTool(
+        default_ontology_prefix=onto_client.default_ontology,
+        onto_lookup=_get_project_and_knora_ontology_from_server(onto_client),
+        save_location=onto_client.save_location,
+    )
+    _iterate_over_xml_find_problems(root, onto_tool)
+
+
+def _get_project_and_knora_ontology_from_server(onto_client: OntologyClient) -> dict[str, Ontology]:
```

Comment:
Generally I agree, but here what we get from the API is an onology called knora-api. We have to deal with this in the strings and json. So in this case I think it would be more confusing